### PR TITLE
docs(storybook): synced with-layer stories with redesign

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
None.

### Description
The latest Carbon React storybook contains redesigned With Layer stories, this PR introduces a template component in order to sync with those stories in our CWC Storybook. 

### Changelog

**New**

- new storybook template folder where the new `with-layer` component lives in

**Changed**

- changes "With layer" stories for Select, Slider, TextArea, TextInput, and Tile

### CODE REVIEW

**Comment by oliviaflory** [comment]
Holy moly @IgnacioBecerra this was so fast!!! 🚀 💥 

They look great, just one minor tweak: the expandable tiles are extending about 16px outside of their layer bounding box, can we get them inset on the right side?

Otherwise everything is perfect 👌 

Screenshot 2023-04-19 at 1 32 47 PM

**Comment by IgnacioBecerra** [comment]
@kennylam Went back to this and realized there was an easier way to copy elements instead of using `unsafeHTML`. Just clone the child nodes and append them directly to the component while assigning them their proper slot position. 🎉

File: packages/carbon-web-components/src/components/slider/slider-story.ts [hunk 1]
```diff
@@ -66,6 +67,7 @@ export const ControlledSlider = () => {
 export const ControlledSliderWithLayer = () => {
   let value = 87;
   function onClick() {
+    console.log('asdjnasjkdbjasbdjkasbsdk');
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/slider/slider-story.ts:70]: Can remove.

File: packages/carbon-web-components/src/components/tile/tile-story.ts [hunk 2]
```diff
@@ -37,24 +38,20 @@ Default.storyName = 'Default';
 export const DefaultWithLayer = (args) => {
   const { colorScheme } = args?.[`${prefix}-tile`] ?? {};
   return html`
-    <cds-layer>
-      <cds-tile color-scheme="${ifDefined(colorScheme)}">
-        First layer
+    <sb-template-layers>
+      <cds-tile slot="layer-1" color-scheme="${ifDefined(colorScheme)}">
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/tile/tile-story.ts:42]:
```suggestion
      <cds-tile color-scheme="${ifDefined(colorScheme)}">
```

File: packages/carbon-web-components/src/components/tile/tile-story.ts [hunk 3]
```diff
@@ -37,24 +38,20 @@ Default.storyName = 'Default';
...
-      </cds-layer>
-    </cds-layer>
+      <cds-tile slot="layer-2" color-scheme="${ifDefined(colorScheme)}">
+        Default layer
+        <a href="https://example.com">Link</a>
+      </cds-tile>
+      <cds-tile slot="layer-3" color-scheme="${ifDefined(colorScheme)}">
+        Default layer
+        <a href="https://example.com">Link</a>
+      </cds-tile>
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/tile/tile-story.ts:53]: Don't need these anymore, right?

**Review by kennylam — COMMENTED** [review]
@IgnacioBecerra Nice! Just a few small things.

File: packages/carbon-web-components/src/components/slider/slider-story.ts (hunk 1 above)
Comment (IgnacioBecerra) [reply packages/carbon-web-components/src/components/slider/slider-story.ts:70]: Woops 😅

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Review by kennylam — APPROVED** [review]
LGTM!

**Pull request merged**